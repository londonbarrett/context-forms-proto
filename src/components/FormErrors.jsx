import * as React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { withFormContext } from './Forms';

const List = styled.ul`
  list-style-position: inside;
  margin-top: .5rem;
  padding-left: 1rem;
  text-align: left;
`;

const ListItem = styled.li`
  font-size: .7rem;
`;

const FormErrors = ({ errors }) => {
  if (errors && errors.length) {
    const items = errors.map(error => (
      <ListItem key={error}>
        {error}
      </ListItem>
    ));
    return (
      <div>
        <h2>
          Form Errors
        </h2>
        <List>
          {items}
        </List>
      </div>
    );
  }
  return null;
};

FormErrors.propTypes = {
  errors: PropTypes.arrayOf(PropTypes.string),
};

FormErrors.defaultProps = {
  errors: undefined,
};

export default withFormContext(FormErrors);
