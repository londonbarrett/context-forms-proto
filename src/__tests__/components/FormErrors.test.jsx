import React from 'react';
import { render } from 'enzyme';
import FormErrors from '../../components/FormErrors';

it('Renders correctly', () => {
  const wrapper = render(
    <FormErrors
      errors={[
        'error 1',
        'error 2',
        'error 3',
      ]}
    />,
  );
  expect(wrapper).toMatchSnapshot();
});
