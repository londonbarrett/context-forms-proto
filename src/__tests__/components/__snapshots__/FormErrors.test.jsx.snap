// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<div>
  <h2>
    Form Errors
  </h2>
  <ul
    class="sc-bdVaJa iwCcLp"
  >
    <li
      class="sc-bwzfXH eUFeNI"
    >
      error 1
    </li>
    <li
      class="sc-bwzfXH eUFeNI"
    >
      error 2
    </li>
    <li
      class="sc-bwzfXH eUFeNI"
    >
      error 3
    </li>
  </ul>
</div>
`;
